File: 1464. Maximum Product of Two Elements in an Array/soln1.py
# Runtime
# 58ms
# Beats 48.52%of users with Python3

# Memory
# 16.35MB
# Beats 41.49%of users with Python3

class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        maxes = [nums[0], 0]
        if len(nums) == 2:
            return (nums[0] - 1) * (nums[1] - 1)
        for i in range(len(nums)):
            if nums[i] >= maxes[0]:
                maxes[0] = nums[i]
                maxes.sort()
        print(maxes)
        return (maxes[0] - 1) * (maxes[1] - 1)
        

File: 1. Two Sum/soln1.cpp
// Runtime 15ms
// Beats 50.17%

// Memory 15.62MB
// Beats 7.47%

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> umap; 
        if (nums.size() < 3) {
            return {0,1};
        }
        for (int i = 0; i < nums.size(); i++){
            umap[nums[i]] = i;
        }
        for (int i = 0; i < nums.size(); i++){
            if (umap.count(target - nums[i]) == 1 && umap[target-nums[i]] != i){

                return {i,umap[target-nums[i]]};
            }
        }
        return {};
    }
};
